File: src/main/java/com/amigos/hacer/porhacer/models/TaskRepository.java
package com.amigos.hacer.porhacer.models;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface TaskRepository extends JpaRepository<Task, Long> {
    Task findByName(String name);

    //find all tasks by group id
    List<Task> findAllByGroupId(Long id);

}

File: src/main/kotlin/com/amigos/amigos/repositories/TaskRepository.kt
package com.amigos.amigos.repositories

import com.amigos.amigos.models.Task
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*


@Repository
interface TaskRepository : JpaRepository<Task?, Long?> {
    fun findByTask(task: String?): Task?
    override fun findAll(): List<Task>
    override fun findById(id: Long): Optional<Task?>
    //find all incompleted tasks
    fun findByIsCompleted(isCompleted: Boolean?): List<Task>

}